### PR TITLE
Miscellaneous changes to rustbuild and CI.

### PR DESCRIPTION
**Pull request by kennytm**
1. Don't build LLVM when running rust-installer.
2. If toolstate is unchanged, don't push a commit to the toolstate repo.
3. Allow `./x.py build src/librustc_codegen_llvm`
4. Added log to track #50887.

### CODE REVIEW

File: src/bootstrap/tool.rs [hunk 1]
```diff
@@ -343,7 +350,7 @@ tool!(
     Compiletest, "src/tools/compiletest", "compiletest", Mode::ToolTest;
     BuildManifest, "src/tools/build-manifest", "build-manifest", Mode::ToolStd;
     RemoteTestClient, "src/tools/remote-test-client", "remote-test-client", Mode::ToolStd;
-    RustInstaller, "src/tools/rust-installer", "fabricate", Mode::ToolStd;
+    RustInstaller, "src/tools/rust-installer", "fabricate", Mode::ToolStd, llvm_tools = false;
```
Comment (Mark-Simulacrum) [inline src/bootstrap/tool.rs]: What makes RustInstaller special? Can we add a comment?

**Review by Mark-Simulacrum — COMMENTED** [review]
Otherwise looks good.

File: src/bootstrap/tool.rs (hunk 1 above)
Comment (kennytm) [reply src/bootstrap/tool.rs]: @Mark-Simulacrum Added a FIXME in ab5e3e6. The main reason is I've only verified RustInstaller can run without the LLVM binaries in $PATH (as the main purpose is to remove unnecessary work from distcheck).

**Comment by kennytm** [comment]
The LLVM dependency was pulled in by the `self.llvm_bin_path()` call, so it will s actually sufficient.

**Pull request merged**